service/s3/internal/customizations/handle_200_error_test.go

```diff
@@ -0,0 +1,71 @@
+package customizations_test
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/internal/awstesting/unit"
+	"github.com/awslabs/smithy-go"
+
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	"github.com/aws/aws-sdk-go-v2/service/s3/types"
+)
+
+type EndpointResolverFunc func(region string, options s3.ResolverOptions) (aws.Endpoint, error)
+
+func (fn EndpointResolverFunc) ResolveEndpoint(region string, options s3.ResolverOptions) (endpoint aws.Endpoint, err error) {
+	return fn(region, options)
+}
+
+func TestErrorResponseWith200StatusCode(t *testing.T) {
```

Comment:
Would be helpful to have two additional test cases

* No response document received at all, the SDK can use the unexpected eof error for this case. Or some other similar error that might be more descriptive.
* Partial response document received, that should fail with unexpected eof.